### ISSUE TITLE
Regenerate documentation only on merge to master

### ISSUE THREAD

**Issue by aabenoja** [WTW-IM/es-components#149]
The current precommit hook can silently fail to regenerate the docs when the `node_modules` directory has been cleaned prior to commit. While the hook does reduce the previous overhead these silent errors can cause malformed documentation that is only caught when merged into master. We should replace this with some sort of bot to correctly generate and publish new documentation on merge to master.

**Comment by aabenoja** [comment]
@jrios @Darrken Let's use this ticket to also include publishing. The script to do this should be pretty straightforward per [this gist](https://gist.github.com/willprice/e07efd73fb7f13f917ea) and [stackoverflow](https://stackoverflow.com/questions/31338562/travisci-run-after-success-on-a-specific-branch)

```yml
after_success:
  - push_and_publish.sh
```

```sh
#!/bin/bash
if [ "$TRAVIS_BRANCH" == "master" ]; then
  // regenerate docs
  npm run publish_only
fi

**Comment by aabenoja** [comment]
Closed by #161
